### DISCUSSION TITLE
Utm parameters ignored

### DISCUSSION THREAD

**Discussion by andrejj** [plausible/analytics#1056] [Self-Hosted Support] [answered]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
**Describe the bug**
UTM parameters not being collected

**Expected behavior**
When visiting links with utm parameters I'd expect them appear in the dashboard.

**Screenshots**

Events are being sent with paramters:

![image](https://user-images.githubusercontent.com/9829/118809194-96453080-b8aa-11eb-8852-92b1a298a41d.png)

But not showing up:

![image](https://user-images.githubusercontent.com/9829/118809333-c7256580-b8aa-11eb-9aa5-9f811b1e32ba.png)

**Environment (If applicable):**
 - Self Hosted
 - OS: Linux
 - Browser Chrome
 - Version Latest

**Comment by ukutaht** [comment, marked as answer]
UTM parameters only show up in the UI if they start a session. Try hitting that page with UTM tags in a pageview that's the first in a session. You can start a new session by:
* Waiting 30 minutes
* Using a different browser
* Using a browser extension to manipulate the `X-Forwarded-For` header in order to change the IP address that Plausible uses.

**Reply by andrejj** [reply]
Ok, I retried after few hours and they appeared. Thank you, this solved it.
Maybe mentioning this in the docs would make sense.

**Reply by superafroman** [reply]
What's the reasoning for this behaviour? 

If a user visits my site, has a look around, maybe signs up to a newsletter and then I email them, upselling them on a product I think they're interested in (don't worry I don't do this...). I'd want the link in the email to have UTM params and for those to be tracked so I could see my nefarious actions were working. But with this logic if they come back within 30mins it won't be tracked.

**Reply by metmarkosaric** [reply]
The fact that the referral sources are counted only when they start a new session on your site prevents external domains such as a payment gateway that the user is being taken through to show up in the referrers list.

**Reply by superafroman** [reply]
But a payment gateway wouldn't set UTM params in order to redirect back to your site.
